src/ProbabilisticProgramming/Loops.dfy

```diff
@@ -93,13 +93,15 @@ module Loops {
   // Definition of until loops (rejection sampling).
   // For proofs, use the lemma `UntilUnroll`.
   // Definition 44
-  ghost function Until<A>(proposal: Monad.Hurd<A>, accept: A -> bool): (f: Monad.Hurd<A>)
+  ghost function Until<A(!new)>(proposal: Monad.Hurd<A>, accept: A -> bool): (f: Monad.Hurd<A>)
     requires UntilTerminatesAlmostSurely(proposal, accept)
     ensures
       var reject := (a: A) => !accept(a);
       var body := (a: A) => proposal;
       forall s :: f(s) == proposal(s).Bind(init => While(reject, body, init))
+    ensures forall s :: f(s).Satisfies(accept)
```

Comment:
That postcondition is wrong as far as I can tell. It's only true that it satisfies the acceptance condition with probability 1.